### ISSUE TITLE
No support for using the CLI/jar checker to validate documents against arbitrary schemas? (e.g., DocBook)

### ISSUE THREAD

**Issue by frankarensmeier** [validator/validator#1823]
I am trying to validate a simple document in the terminal (with the nu.validator.client.SimpleCommandLineValidator). Passing a schema URL starting with https always fails.

Here is a simple input document:

```xml
<?xml version="1.0" encoding="utf-8"?>
<section xmlns="http://docbook.org/ns/docbook" version="5.0" xml:lang="en">
    <title>Title</title>
    <para>Some text</para>
</section>
```

And the command:
```bash
java -cp vnu.jar nu.validator.client.SimpleCommandLineValidator \
    --schema https://docbook.org/xml/5.1/rng/docbook.rng \
    --xml \
    --stdout \
    --format json \
    file.xml
```

This results in **error: The "--schema" option requires a URL for a schema.**

Looking at the source code, it seems that this restriction is hard coded.
 https://github.com/validator/validator/blob/5cd4ab85fb7d175540febd82a71b7a03079eed3f/src/nu/validator/client/SimpleCommandLineValidator.java#L269

Running the Vnu validator as a web service, you actually can specify a schema with https. Is there any particular reason for this restriction? http is discouraged for many years now. And I don't see any workaround at this moment (besides providing some kind of proxy from localhost).

Any help would be greatly appreciated. Thanks!

**Comment by sideshowbarker** [comment]
> I'm not sure custom schemas work at all. I have setup a test server, no https, only http. The schema file is not even downloaded. The only error I get is `Failed to resolve schema URL "http://x.x.x.x.x/docbookxi-5.1.rng". Terminating.`

Resolving schemas is something that happens outside the validator code. And the way that it works, I think it’s expected at that the schema file won’t be downloaded. I believe instead what needs to happen is that you have a local “XML catalog” on your system and that maps the schema URL to a local file path.

And for that to work, you need to specify the `XML_CATALOG_FILES` environment variable in your environment when you run the validator jar.

See https://sagehill.net/docbookxsl/Catalogs.html

But all that said, when I test it locally with the validator jar, it doesn’t seem to be finding the docbook schema file as expected.

So… I dunno — maybe the validator jar doesn’t actually support supplying schema URLs other than the ones it has hardcoded for HTML.

Maybe there’s somebody else who’s watching this issue tracker who has successfully use the validator jar to run checks against non-HTML schemas — and they’ll see this issue and weigh in on it.

I’m sorry I can’t personally be of much further help on this.

**Comment by sideshowbarker** [comment]
@frankarensmeier I finally got around to getting this investigated and fixed.

Your original invocation in the issue description should now “just work”.

Please try it — and if you find anything that’s still not working as expected, please raise issues.
